---
layout: default
---

{% include header.md %}

# Conference programme

The conference will take place on 14-18 December 2020, from 14:00 to 20:00 CET.

### [Bird's eye view of the schedule](images/birdseye.pdf)

## Monday 14 December

| Time | Speaker |
| ---- | ------- |
| 14:00 - 14:10 | Welcome and opening remarks |
| 14:10 - 14:30 | **Martin Morgan - The Bioconductor Project: Updates and Directions** |
| 14:30 - 14:45 | Kevin Rue-Albrecht - BiocChallenges presentation |
| 14:45 - 15:30 | **Keynote: Holger Heyn - A spatially-resolved tumor immune cell atlas for precision immuno-oncology** |
| 15:30 - 15:40 | Break |
| 15:40 - 16:05 | Simone Tiberi - distinct: a novel approach to differential analyses |
| 16:05 - 16:15 | Break |
| 16:15 - 17:00 | **Keynote: Sehyun Oh - Bioinformatics On Cloud: How to leverage cloud-based resources for your bioinformatics works** |
| 17:00 - 17:15 | Break |
| 17:15 - 17:25 | Pierre-Luc Germain - Doublet identification and characterization in single-cell data with scDblFinder |
| 17:25 - 17:35 | Andrea Sottosanti - A new statistical approach for the simultaneous clustering of genes and cells in spatial transcriptomic experiments |
| 17:35 - 17:45 | Kwame Forbes - Integration of preprocessed single-cell datasets with bulk differential analysis results for DESeq2 |
| 17:45 - 17:55 | Brenda Pardo - Using VisiumExperiment objects at spatialLIBD package |
| 17:55 - 18:05 | Break |
| 18:05 - 18:15 | Federico Marini - annoFuse and shinyFuse: Annotating, prioritizing, and interactively exploring putative oncogenic RNA fusions |
| 18:15 - 18:25 | Rory Stark - Normalizing ChIP and ATAC sequencing data for differential analysis |
| 18:25 - 18:35 | Reka Toth - Methrix: a package for systematic aggregation and analysis of bisulfite sequencing data |
| 18:35 - 18:45 | Ricardo Omar Ramirez Flores - Decoupling statistics and networks: a crowdsourced systematic assessment of transcription factor activity estimation from transcriptomics data |
| 18:45 - 19:00 | Break |
| 19:00 - 20:00 | Birds of a feather 1:  Infrastructure to enable spatially resolved transcriptomics data analysis with Bioconductor (Dario Righelli) |
|  | Birds of a feather 2:  Creating Github Actions for building and checking Bioconductor Packages (Mike Smith) |

## Tuesday 15 December

| Time | Speaker |
| ---- | ------- |
| 14:00 - 14:45 | **Keynote: Britta Velten - Integrative analysis of multi-omics data** |
| 14:45 - 14:55 | Break |
| 14:55 - 15:20 | Angela Andreella - pARI package: valid double-dipping via permutation-based All Resolutions Inference |
| 15:20 - 15:30 | Break |
| 15:30 - 15:40 | Thomas Schwarzl - DEWSeq: a new data analysis package for eCLIP data showing improved RNA-binding site assignments for the ENCODE eCLIP data |
| 15:40 - 15:50 | David Zhang - dasper: Detection of aberrant splicing events from RNA-sequencing data |
| 15:50 - 16:00 | Ying Chen - bambu: reference-guided transcript discovery and quantification for long read RNA-Seq data |
| 16:00 - 16:10 | Mattia Furlan - Inference of transcriptional and post-transcriptional dynamics from sequencing experiments |
| 16:10 - 16:20 | Louise Huuki - Correcting for Cell RNA Fractions in MDD RNA-seq Data |
| 16:20 - 16:30 | Break |
| 16:30 - 18:00 | Long Workshop 1: Stefano Mangiola - tidytranscriptomics : introduction to tidy analysis of single-cell and bulk RNA sequencing data |
|  | Long Workshop 2: Rory Stark - Quantitiative analysis of ChiP-seq, ATAC-seq, and related DNA enrichment assays |
| 18:00 - 18:15 | Break |
| 18:15 - 19:15 | [Poster Session](#posters) |
| 19:15 - 20:00 | Short Workshop 1: Marcel Ramos - Multi-omic Integration and Analysis of cBioPortal and TCGA data with MultiAssayExperiment |
|  |  Short Workshop 2: Federico Agostinis - NewWave, new R package for dimensional reduction and batch effect removal for single cell RNA-seq data |

## Wednesday 16 December

| Time | Speaker |
| ---- | ------- |
| 14:00 - 14:15 | Announcement |
| 14:15 - 15:00 | **Keynote: Monica Chiogna - Connections matter: a conversation between statistics and biology** |
| 15:00 - 15:10 | Break |
| 15:10 - 15:35 | Alejandro Reyes - Large-Scale Topological Changes Restrain Malignant Progression in Colorectal Cancer |
| 15:35 - 16:00 | Elyas Heidari - scGCN: A Geometric Deep Learning Framework on Single-cell Gene Networks |
| 16:00 - 16:10 | Break |
| 16:10 - 16:20 | Elisa Salviato - SourceSet: a graphical model approach to identify primary genes in perturbed biological pathways |
| 16:20 - 16:30 | Christian Holland - Estimation of transcription factor and pathway activities from bulk and single-cell transcriptomics data with dorothea and progeny |
| 16:30 - 16:40 | Alejandro Cisterna García - Switching between mendelian genes space and clinical phenotypes space contributes to comparative Mendelian gene sets analysis |
| 16:40 - 16:50 | Leo Lahti - MicrobiomeExperiment: a new class for microbiome data |
| 16:50 - 17:00 | Valeria Policastro - ROBIN (ROBustness In Network): for Comparison and Validation of communities |
| 17:00 - 17:10 | Matteo Calgaro - Assessment of statistical methods from single cell, bulk RNA-seq, and metagenomics applied to microbiome data |
| 17:10 - 17:20 | Riccardo Corradin - BNPmix: an R package to estimate Bayesian nonparametric mixtures |
| 17:20 - 17:30 | Break |
| 17:30 - 18:15 | Short Workshop 3: Spiro C Stilianoudakis - preciseTAD: a machine-learning framework for predicting boundaries of 3D genomic elements |
|  |  Short Workshop 4: Dario Righelli - Spatial Transcriptomics workflow using SpatialExperiment package classes |
| 18:15 - 18:30 | Break |
| 18:30 - 20:00 | Long Workshop 3: Nicholas J Eagles - SPEAQeasy: a Scalable Pipeline for Expression Analysis and Quantification for R/Bioconductor-powered RNA-seq analyses |
|  | Long Workshop 4: iSEE: Interactive SummarizedExperiment Explorer |

## Thursday 17 December

| Time | Speaker |
| ---- | ------- |
| 13:30 - 14:00 | After-lunch informal coffee break |
| 14:00 - 14:45 | **Keynote: Nicola Segata** |
| 14:45 - 14:55 | Break |
| 14:55 - 15:20 | Etienne Becht - Infinity Flow: High-throughput single-cell quantification of 100s of proteins using conventional flow cytometry and machine learning |
| 15:20 - 15:30 | Break |
| 15:30 - 16:15 | **Keynote: Elsa Bernard - Population Genomics and Personalized Prognosis in Myelodysplastic Syndromes** |
| 16:15 - 16:25 | Break |
| 16:25 - 16:50 | Abby Spangler - Evaluation of cell populations in the locus coeruleus through spatial transcriptomics |
| 16:50 - 17:00 | Mike Smith - Browsing and searching the Bioconductor codebase |
| 17:00 - 17:10 | Charlotte Soneson - The Bioconductor teaching committee: A collaborative effort to consolidate Bioconductor-focused training material and establish a community of trainers |
| 17:10 - 17:20 | Naji Faris - "Empower" the Biologist and "liberate" the Bioinformatician |
| 17:20 - 17:30 | Break |
| 17:30 - 18:15 | Short Workshop 5: Laurent Gatto - QFeatures: Quantitative features for mass spectrometry data |
|  |  Short Workshop 6: Mahmoud Ahmed - Integration of ChIP-seq and RNA-seq data in R |
| 18:15 - 18:30 | Break |
| 18:30 - 19:15 | Short Workshop 7: Johannes Rainer - Seamless Integration of Mass Spectrometry Data from Different Sources with the Spectra Package |
| 18:30 - 20:00 | Long Workshop 5: Hector Roux de Bezieux - Trajectory inference across conditions: differential expression and differential progression |


## Friday 18 December

| Time | Speaker |
| ---- | ------- |
| 13:30 - 14:00 | After-lunch informal coffee break |
| 14:00 - 14:45 | **Keynote: Luke Zappia - Interoperability between Bioconductor and Python for scRNA-seq analysis** |
| 14:45 - 14:55 | Break |
| 14:55 - 15:05 | Chen Meng - Interactive and comprehensive exploration of ExpressionSet using ExpressionSetViewer |
| 15:05 - 15:15 | Alan Brian O'Callaghan - Scalable or density-preserving (but not both) non-linear dimensionality reduction with snifter and densvis |
| 15:15 - 15:25 | Maria Dermit - Peptide Correlation Analysis (PeCorA) Reveals Differential Proteoform Regulation |
| 15:25 - 15:35 | Witold Wolski - Prolfqua - R package for proteomics label-free quantification |
| 15:35 - 15:45 | Anne-Maud Ferreira - CyTOF data analysis workflow with Bioconductor packages |
| 15:45 - 15:55 | Pol Castellano Escuder - POMA: An user-friendly workflow for pre-processing and statistical analysis of mass spectrometry data |
| 15:55 - 16:05 | Break |
| 16:05 - 16:30 | Christophe Vanderaa - Standardised and reproducible analysis of mass spectrometry-based single-cell proteomics data |
| 16:30 - 16:55 | Oliver Crook - The effect of phosphorylation on the spatial proteome |
| 16:55 - 17:05 | Break |
| 17:05 - 18:05 | Birds of a feather 3: The emerging R ecosystem for microbiome research (Leo Lahti) |
|  | Birds of a feather 4: Ten simple rules for thriving in bioinformatics research (Federico Marini) |
| 18:05 - 18:20 | Break |
| 18:20 - 19:05 | **Keynote: Henrik Bengtsson** |
| 19:05 - 19:15 | Kevin Rue-Albrecht - BiocChallenges closing |
| 19:15 - 19:20 | Closing remarks |
| 19:20 - 19:25 | Announcements |

## Posters

- Poster 1: Constantin Ahlmann-Eltze - Interactions between explicit and latent covariates detect cell-type specific treatment effects in exploratory single cell analysis without clustering
- Poster 2: Nils Kurzawa - Bioconductor packages for analysis of thermal proteome profiling data
- Poster 3: Marcel Ramos - HCAMatrixBrowser: Making use of the HCA Matrix API in Bioconductor
- Poster 4: Anna Diacofotakis - Activation of digestive-tract and stratified-epithelium transcriptional programs by promoter DNA demethylation in human tumor cells
- Poster 5: Sonia García Ruiz - CoExp Web Application, a web application for the interactive use of co-expression networks
- Poster 6: Kelly Eckenrode - SingleCellMultiModal: Integrating Multi-modal Single Cell Experiment datasets
- Poster 7: Sehyun Oh - GenomicSuperSignature: interpretation of RNA-seq experiments through robust, efficient comparison to public databases
- Poster 8: Leo Lahti - MicrobiomeExperiment: a new class for microbiome data
- Poster 9: Ying Chen - bambu: reference-guided transcript discovery and quantification for long read RNA-Seq data
- Poster 10: Witold Eryk Wolski, Christian Panse - Prolfqua: R package for proteomics label-free quantification
